packages/mui-joy/src/Paper/Paper.tsx

```diff
@@ -0,0 +1,168 @@
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      ownerState.variant === 'elevation' && styles[`elevation${ownerState.elevation}`],
+    ];
+  },
+})<{ ownerState: PaperProps & { mode?: Mode } }>(({ theme, ownerState }) => {
+  return [
+    {
+      // TODO: discuss the theme transition.
+      // This value is copied from mui-material Paper.
+      transition: 'box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+      ...(ownerState.elevation && {
+        boxShadow: theme.shadow[ownerState.elevation],
+        ...(ownerState.mode === 'dark' && {
+          backgroundImage: `linear-gradient(${alpha(
+            '#fff',
+            getOverlayAlpha(ownerState.elevation),
+          )}, ${alpha('#fff', getOverlayAlpha(ownerState.elevation))})`,
+        }),
+      }),
+      ...(!ownerState.variant &&
+        ownerState.color && {
+          backgroundColor: theme.vars.palette[ownerState.color]?.textColor,
+        }),
+      borderRadius: theme.vars.radius.lg,
+    },
+    ownerState.variant && theme.variants[ownerState.variant]?.[ownerState.color!],
+  ];
+});
+
+const Paper = React.forwardRef(function Paper(inProps, ref) {
+  const props = useThemeProps<typeof inProps & PaperProps>({
+    props: inProps,
+    name: 'MuiPaper',
+  });
+
+  const { mode } = useColorScheme();
+  const { className, color = 'neutral', component = 'div', variant, elevation, ...other } = props;
+
+  const ownerState = {
+    ...props,
+    color,
+    component,
+    elevation,
+    variant,
+    mode,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  return (
+    <PaperRoot
+      as={component}
+      ownerState={ownerState}
+      className={clsx(classes.root, className)}
+      ref={ref}
+      {...other}
+    />
+  );
+}) as OverridableComponent<PaperTypeMap>;
+
+Paper.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * The content of the component.
+   */
+  children: PropTypes.node,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * The color of the component. It supports those theme colors that make sense for this component.
+   * @default 'neutral'
+   */
+  color: PropTypes.oneOf(['danger', 'info', 'neutral', 'primary', 'success', 'warning']),
+  /**
+   * The component used for the root node.
+   * Either a string to use a HTML element or a component.
+   */
+  component: PropTypes.elementType,
+  /**
+   * Shadow depth, corresponds to `dp` in the spec.
+   * It accepts theme values between 'xs' and 'xl'.
+   */
+  elevation: PropTypes.oneOf(['lg', 'md', 'sm', 'xl', 'xs']),
+  /**
+   * The variant to use.
+   */
+  variant: PropTypes.oneOf(['contained', 'light', 'outlined', 'text']),
```

Comment:
It should be this:
```suggestion
  variant: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
    PropTypes.oneOf(['contained', 'light', 'outlined', 'text']),
    PropTypes.string,
  ]),
```
otherwise, when developers extends the variant, they will see Proptypes error. also applies to `colors` and `elevation`